### PR TITLE
Re-wording, missing commas, and markdown improvements

### PR DESCRIPTION
**Pull request by Youssef1313**

### CODE REVIEW

File: docs/csharp/programming-guide/generics/index.md [hunk 1]
```diff
@@ -8,66 +8,52 @@ helpviewer_keywords:
...
+[!code-csharp[csProgGuideGenerics#3](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideGenerics/CS/Generics.cs#3)]
+
+## Generics overview
+
+- Use generic types to maximize code reuse, type safety, and performance.
+- The most common use of generics is to create collection classes.
+- The .NET Framework class library contains several new generic collection classes in the <xref:System.Collections.Generic> namespace. These should be used whenever possible instead of classes such as <xref:System.Collections.ArrayList> in the <xref:System.Collections> namespace.
+- You can create your own generic interfaces, classes, methods, events, and delegates.
+- Generic classes may be constrained to enable access to methods on particular data types.
+- Information on the types that are used in a generic data type may be obtained at run-time by using reflection.
```
Comment (pkulikov) [inline docs/csharp/programming-guide/generics/index.md]: I think that might be simpler to go through this list when the items are separated by an empty line, it's more spacious then. However it's subjective. @mairaw are there any guidelines when to insert an empty line between list items?

File: docs/csharp/programming-guide/generics/index.md [hunk 2]
```diff
@@ -8,66 +8,52 @@ helpviewer_keywords:
...
-  
-- [Generics and Reflection](./generics-and-reflection.md)  
-  
-- [Generics in the Run Time](./generics-in-the-run-time.md)  
-  
-## C# Language Specification  
- For more information, see the [C# Language Specification](~/_csharplang/spec/types.md#constructed-types).  
-  
+
+Generics were added to version 2.0 of the C# language and the common language runtime (CLR). Generics introduce the concept of type parameters to the .NET Framework, which make it possible to design classes and methods that defer the specification of one or more types until the class or method is declared and instantiated by client code. For example, by using a generic type parameter `T`, you can write a single class that other client code can use without incurring the cost or risk of runtime casts or boxing operations, as shown here:
```
Comment (BillWagner) [inline docs/csharp/programming-guide/generics/index.md]: I'd remove the first sentence since .NET Framework 2.0 is more than a decade old:

```suggestion
Generics introduce the concept of type parameters to the .NET Framework, which make it possible to design classes and methods that defer the specification of one or more types until the class or method is declared and instantiated by client code. For example, by using a generic type parameter `T`, you can write a single class that other client code can use without incurring the cost or risk of runtime casts or boxing operations, as shown here:
```

File: docs/csharp/programming-guide/generics/index.md [hunk 3]
```diff
@@ -8,66 +8,52 @@ helpviewer_keywords:
...
+
+[!code-csharp[csProgGuideGenerics#1](~/samples/snippets/csharp/VS_Snippets_VBCSharp/csProgGuideGenerics/CS/Generics.cs#1)]
+
+Generic classes and methods combine reusability, type safety, and efficiency in a way that their non-generic counterparts cannot. Generics are most frequently used with collections and the methods that operate on them. Version 2.0 of the .NET Framework class library provides a new namespace, <xref:System.Collections.Generic>, which contains several new generic-based collection classes. It is recommended that all applications that target the .NET Framework 2.0 and later use the new generic collection classes instead of the older non-generic counterparts such as <xref:System.Collections.ArrayList>. For more information, see [Generics in .NET](../../../standard/generics/index.md).
```
Comment (BillWagner) [inline docs/csharp/programming-guide/generics/index.md]: Again, I'd rephrase this more strongly:

```suggestion
Generic classes and methods combine reusability, type safety, and efficiency in a way that their non-generic counterparts cannot. Generics are most frequently used with collections and the methods that operate on them. The <xref:System.Collections.Generic> names contains several generic-based collection classes. The non-generic collections, such as <xref:Systems.Collections.ArrayList> are not recommended and are maintained for compatibility purposes. For more information, see [Generics in .NET](../../../standard/generics/index.md).
```

File: docs/csharp/programming-guide/generics/index.md (hunk 1 above)
Comment (BillWagner) [reply docs/csharp/programming-guide/generics/index.md]: We generally do not leave blank lines between items in a list. This is fine.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks for making these updates @Youssef1313 

I like the changes.

As long as we're updating this article, I'd consider the updates I added.  Also, because this updates the entire article, I'd recommend updating the `ms.date` metadata since this has been updated.

File: docs/csharp/programming-guide/generics/index.md (hunk 3 above)
Comment (Youssef1313) [reply docs/csharp/programming-guide/generics/index.md]: I think you mean `The Generic namespace contains .....` not `The Generic names contains .....`. ?

Comment (BillWagner) [reply docs/csharp/programming-guide/generics/index.md]: Yes, it should be "namespace"

**Review by BillWagner — APPROVED** [review]
